File: medplum-java/medplum-model/src/main/java/com/fhir/PlanDefinition_Goal.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutablePlanDefinition_Goal.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface PlanDefinition_Goal {
  Optional<String> id();

  Optional<List<RelatedArtifact>> documentation();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> start();

  Optional<List<CodeableConcept>> addresses();

  Optional<CodeableConcept> category();

  CodeableConcept description();

  Optional<List<Extension>> extension();

  Optional<List<PlanDefinition_Target>> target();

  Optional<CodeableConcept> priority();

  static ImmutablePlanDefinition_Goal.DescriptionBuildStage builder() {
    return ImmutablePlanDefinition_Goal.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/PlanDefinition_Goal.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { PlanDefinition_Target } from "./PlanDefinition_Target";
import { RelatedArtifact } from "./RelatedArtifact";

export interface PlanDefinition_Goal {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    category: CodeableConcept | null | undefined;
    description: CodeableConcept;
    priority: CodeableConcept | null | undefined;
    start: CodeableConcept | null | undefined;
    addresses: CodeableConcept[] | null | undefined;
    documentation: RelatedArtifact[] | null | undefined;
    target: PlanDefinition_Target[] | null | undefined;
}
